rustsec/src/advisory/metadata.rs

```diff
@@ -81,4 +81,30 @@ pub struct Metadata {
     /// This can be used to soft-delete advisories which were filed in error.
     #[serde(default)]
     pub withdrawn: Option<Date>,
+
+    /// Origin of the advisory content, in case its content was imported from another source.
+    #[serde(default)]
+    pub syndicated_from: Option<Origin>,
+}
+
+/// Where the advisory comes from, if imported from an external source
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(from = "String")]
+pub enum Origin {
+    /// This advisory was imported from the GitHub Advisory Database
+    ///
+    /// In this case, it is required to have a GHSA alias.
+    #[serde(rename = "GHSA")]
+    GithubAdvisoryDatabase,
+    /// Other origin.
+    Other(String),
```

Comment:
How about actually naming the variant `Ghsa` since that maps directly to the prefix of the original advisory ID?

And as it were, there's already an enum for that, although it doesn't quite meet the requirements needed here: https://docs.rs/rustsec/latest/rustsec/advisory/enum.IdKind.html

Reply:
It can be case insensitive. All abbreviations are uppercase and enforcing weird capitalisation just creates confusion.

Reply:
> And as it were, there's already an enum for that, although it doesn't quite meet the requirements needed here: https://docs.rs/rustsec/latest/rustsec/advisory/enum.IdKind.html

I'm not sure it is semantically the same, there is a difference between the Id issuers and the content source. For example CVE ids are issued by MITRE but a part of the information (cvss scoring, etc.) actually come from nvd (or other sources).